src/V3Undriven.cpp

```diff
@@ -381,7 +390,21 @@ class UndrivenVisitor final : public VNVisitor {
                     UINFO(9, " Full bus.  Entryp=" << cvtToHex(entryp) << endl);
                     warnAlwCombOrder(nodep);
                 }
+                if (m_alwaysCombp && entryp->isDrivenWhole() && !entryp->isDrivenAlwaysCombWhole()) {
+                    nodep->v3warn(MULTIDRIVEN,
```

Comment:
What prevents this from firing with e.g.
```
  always_comb begin
       a = 1;
       a = 2;
  end
```

(Add this to test)

I think you need to track if in same NodeAlways block or not.

Reply:
So I did actually consider this case, which is why I have the `&& !entryp->isDrivenAlwaysCombWhole` condition. Basically I'm saying: if we're assigning it within an always_comb and it previously was assigned _outside_ an always_comb, then throw a warning. And then my other warning is the opposite: if we're assigning it outside an always_comb and it was previously was assigned _within_ an always_comb. 

So I don't believe any other tracking is necessary. I'll confirm this however, and regardless I agree that adding your example to the test is a great idea

Reply:
So then how does this get detected with your proper warning (versus a regular MULTIDRIVEN)?

always_comb a = 1;
always_comb a = 2;

Reply:
(As above put it in a test - that's the proof ;)

Reply:
Ah, two separate always_comb blocks driving the same signal isn't something I considered. I'll fix to account for that case. Thank you!